frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1264,6 +1264,37 @@ const IR::Node* TypeInference::postorder(IR::Type_Package* decl) {
 }
 
 const IR::Node* TypeInference::postorder(IR::Type_Specialized *type) {
+    // If there is no template_parameter in constructor parameters
+    // then compiler should reject instantiation
+    // because it cannot infer the type of template_parameter
+    for (auto arg : *(type->arguments)) {
+        if (arg->is<IR::Type_Dontcare>()) {
+            auto decl = refMap->getDeclaration(type->baseType->path);
+            if (auto extern_decl = decl->to<IR::Type_Extern>()) {
+                for (auto template_parameter : extern_decl->typeParameters->parameters) {
+                    for (auto meth : extern_decl->methods) {
+                        if (meth->type->returnType == nullptr) {
```

Comment:
An extern can have multiple constructors. You stop at the first one. It may not be the right one.

Reply:
Working on fix. Thanks!